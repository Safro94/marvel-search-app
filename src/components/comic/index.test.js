import React from 'react';
import { render, wait } from '@testing-library/react';

import Comic from './';

jest.mock('../../components/shared/image', () => 'image');

jest.mock('./style', () => {
  return {
    Comic: 'comic',
    Information: 'information',
    Title: 'title',
    Text: 'text',
    Image: 'image',
  };
});

describe('<Comic /> ', () => {
  beforeEach(() => jest.clearAllMocks());
  const comic = {
    thumbnail: { path: 'path', extension: 'jpg' },
    title: 'Spider-man',
  };

  it('It should render with a description', async () => {
    // Act
    const { container } = render(<Comic comic={comic} />);
    await wait();

    // Assert
    expect(container).toMatchSnapshot();
  });

  it('It should render without a description', async () => {
    //Arrange
    const comicWithDescription = { ...comic, description: '<h1>Lorem ipsum dolor sit amet consectetur,</h1>' };

    // Act
    const { container } = render(<Comic comic={comicWithDescription} />);
    await wait();

    // Assert
    expect(container).toMatchSnapshot();
  });
});
