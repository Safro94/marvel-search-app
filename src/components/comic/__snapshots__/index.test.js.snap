// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Comic />  It should render with a description 1`] = `
<div>
  <comic>
    <image
      url="path/standard_medium.jpg"
    />
    <information>
      <title>
        Spider-man
      </title>
    </information>
  </comic>
</div>
`;

exports[`<Comic />  It should render without a description 1`] = `
<div>
  <comic>
    <image
      url="path/standard_medium.jpg"
    />
    <information>
      <title>
        Spider-man
      </title>
      <text>
        <h1>
          Lorem ipsum dolor sit amet consectetur,
        </h1>
        ...
      </text>
    </information>
  </comic>
</div>
`;
